src/open_producten/utils/filters.py

```diff
@@ -66,3 +105,120 @@ def filter(self, qs, value):
 
         qs = self.get_method(qs)(**{lookup: value, language_lookup: language_code})
         return qs
+
+
+class Operators(models.TextChoices):
+    exact = "exact", _("gelijk aan")
+    gt = "gt", _("grooter dan")
+    gte = "gte", _("grooter dan of gelijk aan")
+    lt = "lt", _("kleiner dan")
+    lte = "lte", _("kleiner dan of gelijk aan")
+    icontains = "icontains", _("case-insensitive partial match")
+    in_list = "in", _("in een lijst van waarden gescheiden door `|`")
+
+
+def validate_data_attr_value_part(value_part: str, code: str):
+    try:
+        variable, operator, val = value_part.rsplit("__", 2)
+    except ValueError:
+        message = _(
+            "Filter '%(value_part)s' heeft niet de format 'key__operator__waarde'"
+        ) % {"value_part": value_part}
+        raise serializers.ValidationError(message, code=code)
+
+    if operator not in Operators.values:
+        message = _("operator `%(operator)s` is niet bekend/ondersteund") % {
+            "operator": operator
+        }
+        raise serializers.ValidationError(message, code=code)
+
+    if operator not in (
+        Operators.exact,
+        Operators.icontains,
+        Operators.in_list,
+    ) and isinstance(string_to_value(val), str):
+        message = _(
+            "Operator `%(operator)s` ondersteund alleen datums and/or numerieke waarden"
+        ) % {"operator": operator}
+        raise serializers.ValidationError(message, code=code)
+
+
+def validate_data_attr(value: list):
+    code = "invalid-data-attr-query"
+
+    for value_part in value:
+        # check that comma can be only in the value part
+        if "," in value_part.rsplit("__", 1)[0]:
+            message = _(
+                "Filter '%(value_part)s' moet de format 'key__operator__value' hebben, "
+                "komma's kunnen alleen in de `waarde` worden toegevoegd"
+            ) % {"value_part": value_part}
+            raise serializers.ValidationError(message, code=code)
+
+        validate_data_attr_value_part(value_part, code)
+
+
+def string_to_value(value: str) -> str | float | date:
+    if is_number(value):
+        return float(value)
+    elif is_date(value):
+        return date.fromisoformat(value)
+
+    return value
+
+
+def is_date(value: str) -> bool:
+    try:
+        date.fromisoformat(value)
+    except ValueError:
+        return False
+
+    return True
+
+
+def is_number(value: str) -> bool:
+    try:
+        float(value)
+    except ValueError:
+        return False
+
+    return True
+
+
+def filter_data_attr_value_part(
+    value_part: str, field_name: str, queryset: models.QuerySet
+) -> models.QuerySet:
+    """
+    filter one value part for data_attr filters
+    """
+    variable, operator, str_value = value_part.rsplit("__", 2)
+    real_value = string_to_value(str_value)
+
+    if operator == "exact":
```

Comment:
With `match-case`, you avoid repeatedly evaluating `if` conditions, making the execution more direct and efficient , and also more readable, I think
```suggestion
    match operator:
        case Operators.EXACT.value:
```